api/src/main/java/org/pmiops/workbench/cdr/CdrVersionService.java

```diff
@@ -39,4 +70,25 @@ public void setCdrVersion(CdrVersion version) {
     }
     CdrVersionContext.setCdrVersionNoCheckAuthDomain(version);
   }
+
+  /**
+   * Retrieve all the CDR versions visible to users with the specified data access level.
+   * When {@link DataAccessLevel#PROTECTED} is provided, CDR versions for both
+   * {@link DataAccessLevel#REGISTERED} and {@link DataAccessLevel#PROTECTED} are returned.
+   * Note: this relies on {@link User#dataAccessLevel} accurately reflecting that the user is in
+   * the authorization domain that has access to the CDR version BigQuery data sets
+   * with the matching {@link DataAccessLevel} values.
+   * @param dataAccessLevel the data access level of the user
+   * @return a list of {@link CdrVersion} in descending timestamp, data access level order.
+   */
+  public List<CdrVersion> findAuthorizedCdrVersions(DataAccessLevel dataAccessLevel) {
+    ImmutableSet<Short> visibleValues = DATA_ACCESS_LEVEL_TO_VISIBLE_VALUES.get(dataAccessLevel);
+    if (visibleValues == null) {
+      return ImmutableList.of();
+    }
+    return cdrVersionDao.findByDataAccessLevelIn(visibleValues)
+        .stream()
+        .sorted(ORDER_BY_DESCENDING_CREATION_TIME_AND_ACCESS_LEVEL)
+        .collect(Collectors.toList());
```

Comment:
So why do we sort the stream instead of letting jpa do the sort?

Reply:
I went back and forth on this. It won't matter performance-wise. The code is sorta simpler with JPA doing the sorting, but it means a really long ugly method name. :) 

Anyway, I switched it to use JPA for the sort.